R/plotFeatureLoadings.R

```diff
@@ -0,0 +1,404 @@
+#' tse <- agglomerateByPrevalence(tse, rank="Phylum", prevalence=0.99, update.tree = TRUE)
+#' tse <- runPCA(tse, name = "PCA", ncomponents = 2)
+#' plotFeatureLoadings(loadings_matrix, method= "PCA", plot = "Heatmap")
+#' 
+#' # Plotting without tree as a barplot
+#' library(mia)
+#' library(ggtree)
+#' library(scater)
+#' data("GlobalPatterns", package = "mia")
+#' tse <- GlobalPatterns
+#' tse <- logNormCounts(tse)
+#' tse <- agglomerateByPrevalence(tse, rank="Phylum", prevalence=0.99, update.tree = TRUE)
+#' tse <- runPCA(tse, name = "PCA", ncomponents = 2)
+#' plotFeatureLoadings(loadings_matrix, method= "PCA", plot = "Barplot")
+#' 
+#' #' # Plotting without tree as a screeplot
+#' library(mia)
+#' library(ggtree)
+#' library(scater)
+#' data("GlobalPatterns", package = "mia")
+#' tse <- GlobalPatterns
+#' tse <- logNormCounts(tse)
+#' tse <- agglomerateByPrevalence(tse, rank="Phylum", prevalence=0.99, update.tree = TRUE)
+#' tse <- runPCA(tse, name = "PCA", ncomponents = 2)
+#' plotFeatureLoadings(loadings_matrix, method= "PCA", plot = "Screeplot")
+NULL
+
+#' @rdname plotFeatureLoadings
+setGeneric("plotFeatureLoadings", signature = c("object"),
+           function(object, ...)
+             standardGeneric("plotFeatureLoadings"))
+
+
+#' @rdname plotFeatureLoadings
+#' @export 
+setMethod("plotFeatureLoadings", signature = c(object = "TreeSummarizedExperiment"),
+          function(object,
+                   method = "PCA", plot = "Heatmap",
+                   ...){
+            
+            .check_parameters(object,
+                              method = method, plot = plot, ...)   
+            
+            if (method == "PCA") {
+              loadings_matrix <- attr(reducedDim(tse, "PCA"), "rotation")
+              df <- data.frame(loadings_matrix)
+              df[["Feature"]] <- rownames(loadings_matrix)
+              df[["Feature"]] <- factor(df[["Feature"]], levels = df[["Feature"]])
+              df <- df[ order(abs(df[["PC1"]]), decreasing = TRUE)[1:10], ]
+              df <- df[ order(df[["PC1"]]), ]
+              df <- df[ order(abs(df[["PC2"]]), decreasing = TRUE)[1:10], ]
+              df <- df[ order(df[["PC2"]]), ]
+            } else {
+              stop("These methods are yet to be implemented", call. = FALSE)
+            }
+            
+            if (plot == "Tree") {
+              
+              phylo <- rowTree(object)
+              circ <- ggtree(phylo, layout = "circular")
+              df <- rowData(object)
+              color <- randomcoloR::distinctColorPalette(
+                length(
+                  unique(
+                    df$Phylum
+                  )
+                )
+              )
+              df <- data.frame(Class = df$Phylum)
+              rownames(df) <- phylo$tip.label
+              
+              
+              df2 <- data.frame(abs(loadings_matrix))
+              rownames(df2) <- phylo$tip.label
+              
+              
+              p <- gheatmap(
```

Comment:
Is this gheatmap new dependency?

Reply:
gheatmap is from ggtree package that is used in others miaViz functions (e.g. plotTree)

Reply:
Thanks, then it is ok